components/brave_wallet_ui/common/hardware/ledgerjs/eth-ledger-untrusted-transport.ts

```diff
@@ -0,0 +1,162 @@
+
+// EthereumLedgerUntrustedMessagingTransport makes calls to the Ethereum app on a Ledger device
+export class EthereumLedgerUntrustedMessagingTransport extends LedgerUntrustedMessagingTransport {
+  constructor (targetWindow: Window, targetUrl: string) {
+    super(targetWindow, targetUrl)
+    this.addCommandHandler<UnlockResponse>(LedgerCommand.Unlock, this.handleUnlock)
+    this.addCommandHandler<EthGetAccountResponse>(LedgerCommand.GetAccount, this.handleGetAccount)
+    this.addCommandHandler<EthSignTransactionResponse>(LedgerCommand.SignTransaction, this.handleSignTransaction)
+    this.addCommandHandler<EthSignPersonalMessageResponse>(LedgerCommand.SignPersonalMessage, this.handleSignPersonalMessage)
+    this.addCommandHandler<EthSignEip712MessageResponse>(LedgerCommand.SignEip712Message, this.handleSignEip712Message)
+  }
+
+  private handleGetAccount = async (command: EthGetAccountCommand): Promise<EthGetAccountResponse> => {
+    const transport = await TransportWebHID.create()
+    const app = new Eth(transport)
+    try {
+      const result = await app.getAddress(command.path)
+      const getAccountResponsePayload: EthGetAccountResponsePayload = {
+        success: true,
+        publicKey: result.publicKey,
+        address: result.address,
+        chainCode: result.chainCode
+      }
+      const response: EthGetAccountResponse = {
+        id: command.id,
+        command: command.command,
+        payload: getAccountResponsePayload,
+        origin: command.origin
+      }
+      return response
+    } catch (error) {
+      const response: EthGetAccountResponse = {
+        id: command.id,
+        command: command.command,
+        payload: error,
+        origin: command.origin
+      }
+      return response
+    } finally {
+      await transport.close()
+    }
+  }
+
+  private handleSignTransaction = async (command: EthSignTransactionCommand): Promise<EthSignTransactionResponse> => {
+    const transport = await TransportWebHID.create()
+    const app = new Eth(transport)
+    try {
+      const result = await app.signTransaction(command.path, command.rawTxHex)
+      const signTransactionResponsePayload: EthSignTransactionResponsePayload = {
+        success: true,
+        v: result.v,
+        r: result.r,
+        s: result.s
+      }
+      const response: EthSignTransactionResponse = {
+        id: command.id,
+        command: command.command,
+        payload: signTransactionResponsePayload,
+        origin: command.origin
+      }
+      return response
+    } catch (error) {
+      const response: EthSignTransactionResponse = {
+        id: command.id,
+        command: command.command,
+        payload: error,
+        origin: command.origin
+      }
+      return response
+    } finally {
+      await transport.close()
+    }
+  }
+
+  private handleSignPersonalMessage = async (command: EthSignPersonalMessageCommand): Promise<EthSignPersonalMessageResponse> => {
```

Comment:
nit can we rename this `handleSignEip191PersonalMessage` in case it's not clear to others that PersonalMessage -> Eip191 sign format

Reply:
Ah I didn't know that was the standard.  Renaming is fine with me, though right now these names map 1:1 with the function names in the ledger library https://github.com/LedgerHQ/ledger-live/tree/develop/libs/ledgerjs/packages/hw-app-eth#table-of-contents.  Does that impact your thinking at all?

Reply:
Yeah would rather match now that you point that out. Fine to leave as is if that's how they have this since we're inherently tying the implementation to what they're doing and defining